### DISCUSSION TITLE
Delete Rows with Postgres 13

### DISCUSSION THREAD

**Discussion by sudpuzzer** [Budibase/budibase#5217] [Help] [answered]
I am trying to recreate the auto-generated screens for a simple table, but I am having some trouble.

Budibase v1.0.98
Postgres 13.6 Docker image (postgres:13)

I have a simple table "Brands" in Postgres that I created using Budibase:
![image](https://user-images.githubusercontent.com/10662820/161320473-cd892998-a358-4f8b-9541-ec782507d0e7.png)

I have created a Brand Detail page with a 'Delete' button, using the same structure as the auto-generated screens:
![image](https://user-images.githubusercontent.com/10662820/161320601-677e32b9-51ec-4cbd-a0d7-0c689a09e533.png)

Here are the actions on the delete button:
![image](https://user-images.githubusercontent.com/10662820/161320833-ccaab994-6de9-4934-8fc1-dfc8291f7c99.png)
![image](https://user-images.githubusercontent.com/10662820/161321215-94823c34-bb53-4f46-b55d-2508544be7c0.png)

However, when I try to use the delete button, it returns to the brands list as expected, but it does not delete the row. I have tried removing the navigate action and using _id instead of id, but it still does not work. There is no error message that appears on screen or API errors in the developer console in chrome. I am able to write a delete query with a binding to the id column which can be used with the execute query action that works as expected as well. Just can't get the Delete Row action to work.

The autogenerated screen works as expected and gives a Row Deleted confirmation on the screen. The only difference in the auto-generated screens is that the Row Rev field is filled out in the Delete Row action, and it does use _id:
![image](https://user-images.githubusercontent.com/10662820/161322095-987b1168-2956-4331-a070-cec861309359.png)

It is unclear where the Row Rev field is supposed to come from on an external database. Is there something I am missing here?

Thanks!

**Comment by melohagan** [comment]
Hi @sudpuzzer 

**_id** would be the correct field name to use. 

Looking at the gnarly GUID in the Row Rev, the binding has gotten broken somewhere. I would remove both the Row ID and Row Rev, and then re-add the binding for the Row ID.

**Comment by sudpuzzer** [comment, marked as answer]
Hi @melohagan,

I've tried both {{ Brand Repeater.Brands.id }} and {{ Brand Repeater.Brands._id }}, both with the Row Rev blank and copying the Row Rev GUID from the autogenerated screens and none of them work. It doesn't offer any error message, it just does the Navigate action and the row hasn't been deleted.

However, what does work is putting a 0 in the Row Rev field, both with id and _id. Not sure why.

I think the delete action could use some better explanation in the docs, particularly what the Row Rev field is supposed to be for in the case of internal and external databases.

**Reply by melohagan** [reply]
I agree. The Row Rev is specific to CouchDB (the internal budibase db). For external datasources it probably shouldn't be shown at all.
Glad you got it working though!
